### PR TITLE
[nrf fromtree] boot: SHA512 verification

### PR DESCRIPTION
**Pull request by de-nordic**
adds TLV and Kconfig to decouple verification from
other options.

There are two important commits:
[[nrf fromtree] boot: SHA512 verification](https://github.com/nrfconnect/sdk-mcuboot/pull/336/commits/35f1ad114dd6c4e84ece6ef656f877f40b4aa142)
[[nrf fromlist] imgtool: Add support for calculating SHA512](https://github.com/nrfconnect/sdk-mcuboot/pull/336/commits/45b792a778755f1e7effac9baf7fc4a921711695)

all the others commits are revert and bring back of noup commits that cause conflicts and required cherry-picks.

### CODE REVIEW

File: boot/bootutil/include/bootutil/crypto/ecdsa.h [hunk 1]
```diff
@@ -613,6 +619,43 @@ static inline int bootutil_ecdsa_parse_public_key(bootutil_ecdsa_context *ctx,
...
+
+	/* As described on the compact representation in IETF protocols,
+	 * the first byte of the key defines if the ECC points are
+	 * compressed (0x2 or 0x3) or uncompressed (0x4).
+	 * We only support uncompressed keys.
+	 */
+	if (pk[0] != 0x04)
+		return -1;
+
+	pk++;
```
Comment (nordicjm) [inline boot/bootutil/include/bootutil/crypto/ecdsa.h]: wild indent, and use brackets for if

File: boot/bootutil/include/bootutil/crypto/ecdsa.h [hunk 2]
```diff
@@ -622,43 +623,45 @@ static inline int bootutil_ecdsa_parse_public_key(bootutil_ecdsa_context *ctx,
...
-	 * the first byte of the key defines if the ECC points are
-	 * compressed (0x2 or 0x3) or uncompressed (0x4).
-	 * We only support uncompressed keys.
-	 */
-	if (pk[0] != 0x04)
-		return -1;
+    /* Only support uncompressed keys. */
+    if (pk[0] != 0x04) {
+        return -1;
+    }
```
Comment (nordicjm) [inline boot/bootutil/include/bootutil/crypto/ecdsa.h]: the changes here can be squashed to original commit/fixup!

Comment (de-nordic) [reply boot/bootutil/include/bootutil/crypto/ecdsa.h]: OK, can do that.

Comment (de-nordic) [reply boot/bootutil/include/bootutil/crypto/ecdsa.h]: Fixed

**Review by nordicjm — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**